packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1043,8 +1127,37 @@ class TextSelectionGestureDetectorBuilder {
   ///    this callback.
   @protected
   void onSingleTapUp(TapUpDetails details) {
+    if (_isShiftTapping) {
```

Comment:
This reminds me, what happens if the user presses <kbd>shift</kbd> and drag?

Reply:
Good call, I didn't even think to try that.  Looks like the drag should extend the selection created by the shift + tapdown.  Unfortunately what Flutter does with this PR is to start a new selection when the drag starts.  I'll merge this PR since it's ready to go, but I'll follow up with another one to handle the drag.  I won't close the issue (https://github.com/flutter/flutter/issues/91979) until both PRs are merged.